### DISCUSSION TITLE
Validation with DeepCwind test data

### DISCUSSION THREAD

**Discussion by SudharsanKalidoss** [OpenFAST/openfast#882] [Q&A]
Hi everyone, 

   I'm trying to replicate the validation results in the NREL 5 MW wind turbine validation study on the floating semi-submersible platform (OC4). In wind only conditions, one case is simulated with dynamic wind. I'm not sure how to define dynamic wind. Do I need to generate a HAWC format wind file to generate dynamic wind or is there some other method? I have attached the article which I'm following for the simulation. I'm trying to replicate the condition in section IV.C

Thanks

Sudharsan

[Validation of a FAST semi-submersible floating wind turbine numerical model with DeepCwind test data.pdf](https://github.com/OpenFAST/openfast/files/6471952/Validation.of.a.FAST.semi-submersible.floating.wind.turbine.numerical.model.with.DeepCwind.test.data.pdf)

**Comment by jjonkman** [comment]
Hi @SudharsanKalidoss,

This paper was written many years ago and I was not directly involved in running the simulations.   But I would guess that TurbSim was used to generate the dynamic wind data for use in FAST with TurbModel = "API", which is the same as the NPD spectrum mentioned in the paper.

The API (NPD) spectrum implemented in TurbSim, in reference to Coulling et al's paper, was discussed a bit on our forum: https://wind.nrel.gov/forum/wind/viewtopic.php?f=14&t=1896.

Best regards,

**Comment by SudharsanKalidoss** [comment]
Hi @jjonkman,

     Thanks for the help. I generated the API wind profile using Turbsim v2. 

I tried running the OC4 semisubmersible model with default platform properties. It worked fine. When changed the platform properties to match the values in the above mentioned paper, I'm facing the following error.

FAST_Solution:FAST_AdvanceStates:SolveOption2c_Inp2AD_SrvD:InflowWind_CalcOutput:CalcOutput:IfW_Un
iformWind_CalcOutput:GetWindSpeed:Height must not be negative. Error calculating the wind speed
at position (4.8964, -0.20597, -1.10730E-03) in the wind-file coordinates

I already faced this error before for the monopile configuration. You asked me to increase the lowest to the tower node. I'm not sure how can I change the tower node for the floating platform since the lowest tower node starts from 10m.

**Comment by andrew-platt** [comment]
Hi @SudharsanKalidoss 

How far does the simulation get before this error occurs? Is it possible the revised platform does not have sufficient buoyancy for the weight of the turbine and is sinking?

Regards,

**Comment by SudharsanKalidoss** [comment]
Hi @andrew-platt

The simulation run time varies with the platform mass. The default platform mass is 3.85E6. The error occurs when platform mass is increased above 6E6. For the platform mass of 7E6, the error occurs after 12s of simulation time. For the platform mass of 13.44E6, the error occurs at 2s. 

Thanks

**Comment by andrew-platt** [comment]
Hi @SudharsanKalidoss 

Have you plotted the platform heave motion for these simulations?

**Comment by SudharsanKalidoss** [comment]
Hi @andrew-platt 

  I have attached the platform heave motion.
![Plt_heave_13e6](https://user-images.githubusercontent.com/17265645/120519655-dde6b480-c3a0-11eb-8a06-b6dcf131bd65.jpg)
![Plt_heave_7e6](https://user-images.githubusercontent.com/17265645/120519656-dde6b480-c3a0-11eb-9b1d-f0c7d620017e.jpg)
![Plt_heave_9e6](https://user-images.githubusercontent.com/17265645/120519657-dde6b480-c3a0-11eb-81f9-0bf51f0da5e5.jpg)

**Comment by andrew-platt** [comment]
Hi @SudharsanKalidoss 

It appears your platform doesn't have sufficient buoyancy and sinks until the tower base or blade tip strikes the water surface.  From the above plots, it doesn't look like it will reach an equilibrium position before the entire turbine is underwater. 

You will need to adjust the appropriate values to get neutral buoyancy (pontoon and column diameters, etc).

**Comment by SudharsanKalidoss** [comment]
Hi @andrew-platt 

I assume changing the platform geometry means creating a new pot file for HydroDyn. Am I missing some other location to change the geometry of platform?

**Comment by andrew-platt** [comment]
@SudharsanKalidoss 

Changes in the platform geometry will result in changes to the hydrodynamic RAOs (response amplitude operators) which are contained in the potential flow files.  Those will need to be calculated by an external code such as WAMIT.

However, if you are not concerned with the accuracy of the hydrodynamics, you might be able to use the linear hydrodynamic matrices (`AddF0`, `AddCLin`, `AddBLin`, and `AddBQuad`) to approximate the necessary buoyancy and changes expected to the RAO.  However I would expect the frequencies of the surge, pitch, roll, heave and other responses could be incorrect since the RAO's would not be correct.  Alternatively, you could turn off some of the DOF's for the platform (in the ElastoDyn input file), but again I would not expect the frequencies to be correct.

Perhaps @jjonkman will have some additional comments/corrections when he returns.

**Comment by whutlhy** [comment]
Hi @SudharsanKalidoss 
I'm trying to replicate the validation results too, but I can't generate the NPD (API)  spectrum due to the error "CalcFourierCoeffs:CalcFourierCoeffs_API:Error in Cholesky factorization". I noticed that there used to be someone asked this question in the forum too but didn't solve it. What's your solution for this problem? BTW, the link  https://wind.nrel.gov/forum/wind/viewtopic.php?f=14&t=1896. is not available now, where can I read it now? @jjonkman

**Reply by jjonkman** [reply]
Dear @whutlhy,

Here are a couple topics on our new forum that reference the API (NPD) spectrum within TurbSim:
https://forums.nrel.gov/t/wind-wave-data/207
https://forums.nrel.gov/t/api-wind-spectrum/1673

A Cholesky factorization error is typically resolved by coarsening the TurbSim grid, e.g., as discussed in the following forum topic:
https://forums.nrel.gov/t/wind-veer-parameter/2103

Best regards,
